index.js

```diff
@@ -1,91 +1,158 @@
-  if (!(this instanceof TypeScript)) {
-    return new TypeScript(inputTree);
+  
+  var self = this;
+  this.rootFilenames = [];
+  this.files = {};
+  this.cwd = undefined;
+  
+  this.emitFile = function(fileName, srcDir, outputPath) {
+      var output = self.services.getEmitOutput(fileName);
+      var hasErrors = self.hasErrors(fileName);
+      
+      // Log Errors
+      if (hasErrors) {
+          this.logErrors(fileName, srcDir);
+      }
+      
+      // Write File
+      output.outputFiles.forEach(function (o) {
+        outputPath = o.name.replace(srcDir, outputPath);
+        filendir.writeFileSync(outputPath, o.text, "utf8");
+      });
+      
+      return hasErrors;
   }
-
-  Filter.call(this, inputTree, {extensions: ['ts','d.ts'], targetExtension: 'js'});
-
-  this.options = parseOptions((options && options.tsconfig) || path.join(process.cwd(), "tsconfig.json"));
-
-  this.name = 'broccoli-typescript-compiler';
-}
-
-TypeScript.prototype = Object.create(Filter.prototype);
-TypeScript.prototype.constructor = TypeScript;
-
-/*
- * @private
- *
- * @method optionsString
- * @returns a stringifeid version of the input options
- */
-TypeScript.prototype.optionsHash  = function() {
-  if (!this._optionsHash) {
-    this._optionsHash = crypto.createHash('md5').update(stringify(this.options), 'utf8').digest('hex');
+  
+  this.hasErrors = function(fileName) {
+    return self.services.getCompilerOptionsDiagnostics()
+            .concat(self.services.getSyntacticDiagnostics(fileName))
+            .concat(self.services.getSemanticDiagnostics(fileName)).length > 0;
+  };
+  
+  this.logErrors = function(fileName, srcDir) {
+      var allDiagnostics = self.services.getCompilerOptionsDiagnostics()
+          .concat(self.services.getSyntacticDiagnostics(fileName))
+          .concat(self.services.getSemanticDiagnostics(fileName));
+          
+      allDiagnostics.forEach(function (diagnostic) {
+          var message = ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n");
+          
+          if (diagnostic.file) {
+              var _a = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start), line = _a.line, character = _a.character;
+              console.log(chalk.red(diagnostic.file.fileName.replace(srcDir, "") + " (" + (line + 1) + "," + (character + 1) + "): " + message));
+          }
+          else {
+              console.log(chalk.red(message));
+          }
+      });
   }
-  return this._optionsHash;
-};
-
-TypeScript.prototype.cacheKeyProcessString = function(string, relativePath) {
-  return this.optionsHash() + Filter.prototype.cacheKeyProcessString.call(this, string, relativePath);
-};
-
-TypeScript.prototype.processString = function (string, relativePath) {
-  try{
-    return ts.transpileModule(string, {compilerOptions: this.options, fileName: relativePath}).outputText;
-  }catch(e){
-    console.log("TYPESCRIPT ERROR: " + e.message)
-    console.log(e.stack + "\n");
+  
+   this.servicesHost = {
```

Comment:
seems like this should just be a `function ServiceHost` object, that is locally instantiated.